lib/internal/test_runner/runner.js

```diff
@@ -413,19 +413,22 @@ function runTestFile(path, filesWatcher, opts) {
 function watchFiles(testFiles, opts) {
   const runningProcesses = new SafeMap();
   const runningSubtests = new SafeMap();
-  const watcher = new FilesWatcher({ __proto__: null, debounce: 200, mode: 'filter', signal: opts.signal });
+  const watcherMode = opts.hasFiles ? 'filter' : 'all';
+  const watcher = new FilesWatcher({ __proto__: null, debounce: 200, mode: watcherMode, signal: opts.signal });
+  if (!opts.hasFiles) {
+    watcher.watchPath(process.cwd()); // TODO: https://github.com/nodejs/node/issues/53867 before closing this MR
```

Comment:
Please pass `cwd` as an option to `watchFiles` instead.

Reply:
Sure 😁 
I'm going to update this while adding an input to `run` as per your request 🚀 
> We are trying to use inputs for `run()`. #53867. Do not use `process.cwd()` here, as we might be running tests for a different folder, this should be an input to this function.
> 
> Take a look at #53866.

Reply:
I'm not sure this is actually doable in this PR. One thing at a time.

Reply:
no problem at all then 🚀

Reply:
@mcollina, once this PR lands (if it does), should I work on exposing a new flag to allow users to easily select the watched path and add it as an input to `run`?

Something like `--watch-path ./example/path`?

Reply:
> should I work on exposing a new flag to allow users to easily select the watched path and add it as an input to run?

If you want to add a new CLI flag, I'd suggest you first to create an issue to describe why and add all the refs to it (_make sure you mention that you'd like to work on it to avoid other folks working on it at the same time_) because a CLI flag usually involves more considerations and discussions, IMO it would help you with your implementation.

Reply:
ok, makes sense 🚀 I'll follow your suggestion